File: loops/notes.py
#Maisha paredes, notes python loops

#1. What is a loop? 
    #A section of code that repeats multiple times
#2. What are the two types of loops?
    #for loop
nums = [12, 3, 66, 7, 3, 3, 2]
for num in nums:
    print(num)
    #for loop
x = 0

while x < 10:
    print(x)
    x += 1

#3. What is iteration

    #one instance of the loop
    #one particular instance of the loop#iteration in it of itself is the amount of times your looping through

#4. What are lists? 
nums = [1,2,3,4,5,6,7,6]
siblings = ["Angie", "Hanah", "Josh", "Ethan", "Oliver", "Maisha"] #<==brackets, add in items w correct data types, and a comma between each item

    #a group of many things that are related
    #a variable that holds multiple values
print(nums) #prints whole list is ugly
print(siblings[3])#prints 1 item from the list

siblings[7] = "Maisha.two.point.O"
siblings.pop(5)
siblings.insert(1, "Jayshree")
siblings.insert(2, ["Joe", "Noah", "Zee"])
print(siblings)


#5. How do you make lists in python? 

#6. How do you make for loops in python? 

for sibling in siblings: 
    print(sibling)
    for x in range(0,101,20)
        print(x)
#7. How do you make while loops in python?
import random
x = 1 #variable to keeo vount of iteration
goose = random.randint

while x <= 20:
    if x == goose:
        print("goose!")
       break #tells the loop to stop
    else:
        print("duck")
    print("duck")
    x += 1 
    #continues move on to the next iteration without finishing

File: loops/notes.c
// Maisha Paredes, notes c
#include <stdio.h>

int main(void){

//1.What is a loop? 

    //a section of code repeated multiple times

//2.What are the two types of loops?

    //While loops
    int start = 0;
    while(start <5){
        printf("hello!\n");
        start++;
    }
    //for loops
    int q;
    for(q=0;q<5;q++){
        print("%d\n", q);
    }

//3.What is iteration
    //repeating something with minor changes each line

//4.What are arrays? (lists in c)
    //A list is a variable holding multiple values
     
//8. How do you make arrays? (lists in c)
    //arrays all need to be the same data type
int grades[] = {88, 97, 100, 70, 72, 99, 61};
    //1. set data type first 2. AFTER naming put brackets aand write the length of list 3. list is surroumded by curley brackets, 4. commas between in each item
//print a simgle item from a list
printf("CSP GRADE: %d\n", grades[2]);
//Change item in the array
grades[2] = 95;
printf("CSP grade: %d\n", grades[2]);
//size of list in bytes
int size = sizeof(grades);

//length in list items
int length = sizeof(grades)/sizeof(grades[0]);
printf("the array is %d items long.\n", length);
//array w strings
//first brackets sets the lengh of the array
//second breacket sets the length of each string
char movies[][20] = {"lego movie", "kubo and the two strings", "jojo rabbit", "Avengers endgame"};

prinf("the movie is %s!\n", movies[1]);
int mlength = sizeof(movies)/sizeof(movies[0]);
//9. How do you make for loops in C?
//set iterator, keeps track of times going through loop (best practice set as x or i)
int x;
//in parens (starting point; ending point; count by)
for(x=0;x<=10;x+=2){
    printf("%d\n", x);
}
int m;
for(m=0;m<mlength;m++){
    printf("%s\n", movies[m]);
}
//10. How do you make while loops in C?
int w = 100;//start

while(w<=0){//stop
    printf("%d\n", w);
    w -= 10; //what we count by
}
    return 0;
    }




